server/src/test/kotlin/org/elaastic/player/assignmentview/AssignmentOverviewModelFactoryTest.kt

```diff
@@ -0,0 +1,304 @@
+package org.elaastic.player.assignmentview
+
+import org.elaastic.sequence.ExecutionContext
+import org.elaastic.sequence.State
+import org.elaastic.sequence.interaction.InteractionType
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class AssignmentOverviewModelFactoryTest {
+
+    val chart = AssignmentOverviewModelFactory.chartIcon
+    val lock = AssignmentOverviewModelFactory.lockIcon
+    val minus = AssignmentOverviewModelFactory.minusIcon
+    val comment = AssignmentOverviewModelFactory.commentIcon
+    val comments = AssignmentOverviewModelFactory.commentsIcon
+    
+    fun testOfResolveIcons(
```

Comment:
In case of a function that have many test cases (like ``test resolveIcons when Distance, beforeStart, result aren't publish and is teacher`` (line 164) who have to test for every Interaction type (4) and if the result is published or not (2)), I found this function hepful.

Reply:
Let's try to analyze it.

### With the function
```
testOfResolveIcons(
	listOf(minus),
	true,
	ExecutionContext.Distance,
	State.beforeStart,
	resultsArePublished = false,
	null
)
```

### Without the function
```
assertEquals(
	listOf(minus),
	resolveIcons(
		true,
		ExecutionContext.Distance,
		State.beforeStart,
		resultsArePublished = false,
		null
	)
)

```  What does it change ?
From a number of lines of code perspective: you have compacted 10 lines into 8 lines.
From a conceptual point of view : 
- I have no idea of the role of each parameter of `testOfResolveIcons` at the 1st glance. I have to inspect it to understand what it is testing. I can only deduce it is testing some stuff about icons resolving.
- When I look at `assertEquals(<list of icons>, resolveIcons(<parameters>)`, I have no idea how icons are resolved (I need to inspect `resolveIcons` to understand it, but I clearly understand that I'm testing the icons resolution, and the expected value in this case is the provided list. 

I feel like this a better abstraction decomposition. Did I convince you?

Reply:
I would add that the list of icons is not at the same abstraction level of the resolveIcons parameters.

Reply:
Fix in 9c85f3c5d569640dbea80d6482f76b32f6a742b8